### PR TITLE
Convert lattice cache (networked and offline) into use of capability provider

### PR DESCRIPTION
**Pull request by autodidaddict**
Not yet ready for full review. Starting the PR as a WIP before vacation.

### CODE REVIEW

**Comment by autodidaddict** [comment]
@brooksmtownsend when you go through this PR, can you please ensure that this branch doesn't break any of your REPL demo activities?

**Comment by brooksmtownsend** [comment]
@autodidaddict Sure thing. Is it ready for review now? I'd be happy to run through the REPL demo too before it's ready if you'd like.

**Comment by autodidaddict** [comment]
It's ready for testing via the REPL, yes. Not sure if I'm ready with the code being complete.

**Comment by brooksmtownsend** [comment]
KVCounter example works great with this branch (latest commit #8bb6bc8). I appreciate the link name for the lattice cache, could we do that with extras too? (Here's the host inventory)
```
                       Host Inventory (NCA3W27HJLGSY62V5XKJXZTVMBWMXRAOC2BEWPYK3JUIIH45YGLL4KU4)

 hostcore.os                                                macos
 hostcore.arch                                              x86_64
 hostcore.osfamily                                          unix

 Actor ID                                                   Image Reference
 MC6Z57M3HOUPIAMTZDMKFBJ5AZ3UOWZ4WJYO7DZNA6567XSJNBUQP55I   wascc.azurecr.io/kvcounter:v0.1.1

 Provider ID                                                Link Name                  Image Reference
 VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M   default                    wascc.azurecr.io/httpsrv:v0.9.2
 VAHNM37G4ARHZ3CYHB3L34M6TYQWQR6IZ4QVYC4NYZWTJCJ2LWP7S6Z2   __wasmcloud_lattice_cache  N/A
 VAZVC4RX54J2NVCMCW7BPCAHGGG5XZXDBXFUMDUXGESTMQEJLC3YVZWB   default                    wascc.azurecr.io/redis:v0.9.2
 VDHPKGFKDI34Y4RN4PWWZHRYZ6373HYRSNNEM4UTDLLOGO5B37TSVREP   default                    N/A
```

**Comment by autodidaddict** [comment]
This looks fantastic. We can't use a "hidden" link name for extras because that's something that actors use via the interface (we haven't generated it yet...). In other words, regular actors that want GUIDs or random numbers need to make calls to "wascc:extras" "default".

**Comment by brooksmtownsend** [comment]
Ah okay. I forgot about https://github.com/wasmCloud/wasmCloud/issues/24 actually, which would directly address my concern with someone looking at this random provider and saying "huh, don't recognize that, let's just stop it"

**Comment by autodidaddict** [comment]
Also worth testing - fire up a `wasmCloud` binary in a shell with the `KVCACHE_NATS_URL` environment variable set.
Then do a `wash up` in another shell, also with `KVCACHE_NATS_URL` set. 
Then make sure that all of the "multiplayer" repl commands work properly, assuring that the kv cache replication is working.

I've tested this locally a bit, but not thoroughly since I've been trying to automate some of the other distributed tests (to no avail)

File: crates/wasmcloud-host/src/messagebus/latticecache_client.rs [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+        if gr.exists {
+            Ok(Some(gr.value))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Retrieves the list of all currently known OCI image references. You can this list in a loop to
+    /// retrieve all of the image reference mappings.
+    pub async fn get_oci_references(&self) -> Result<Vec<String>> {
```
Comment (brooksmtownsend) [inline crates/wasmcloud-host/src/messagebus/latticecache_client.rs:134]: Does this comment mean to say that I can iterate through this list to retrieve all of the image reference mappings?

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
A couple places for changes, submitting this before I've run the tests locally as I need to reboot. One more suggestion is that I saw many warnings during compilation, we should consider opening an issue to address cargo / clippy warnings before the 0.15.0 release to clean up the code.

File: crates/wasmcloud-host/src/host_controller/hc_actor.rs [hunk 2]
```diff
@@ -375,55 +514,61 @@ impl Handler<StartActor> for HostController {
...
-                        }
-                        Err(e) => Err(format!("Failed to initialize actor: {}", e).into()),
-                    },
-                    Err(_e) => {
-                        error!("Failed to initialize actor - mailbox error");
-                        Err("Failed to initialize actor - mailbox error".into())
-                    }
-                }),
+            async move {
+                new_actor.send(init).await??;
```
[the remark this answers is not in the document]
Comment (autodidaddict) [reply crates/wasmcloud-host/src/host_controller/hc_actor.rs:523]: they can bubble up. the first `?` catches the "mailbox error" from actix, the second unwraps the Result that was returned by the actor.

File: crates/wasmcloud-host/src/messagebus/utils.rs [hunk 3]
```diff
@@ -70,3 +68,16 @@ pub(crate) fn generate_link_invocation(
 
     t.send(inv)
 }
+
+pub(crate) fn system_actor_claims() -> Claims<wascap::jwt::Actor> {
+    Claims::<wascap::jwt::Actor>::new(
+        SYSTEM_ACTOR.to_string(),
+        "ACOJJN6WUP4ODD75XEBKKTCCUJJCY5ZKQ56XVKYK4BEJWGVAOOQHZMCW".to_string(),
```
[the remark this answers is not in the document]
Comment (autodidaddict) [reply crates/wasmcloud-host/src/messagebus/utils.rs:75]: Yep that's why we can also embed the JWTs for the extras and default kvcache providers.

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**